packages/react/src/components/DatePicker/DatePicker.js

```diff
@@ -29,6 +29,38 @@ l10n.en.weekdays.shorthand.forEach((day, index) => {
   }
 });
 
+// Handle how flatpickr wants to see locale
+const formatLocale = (locale) => {
+  switch (locale) {
+    case 'zh-tw':
+      return 'zh_tw';
+    case 'zh-cn':
+      return 'zh';
+    default:
+      return locale;
+  }
+};
+
+const formatDate = (dateFormat, locale) => {
+  if (dateFormat !== undefined) return dateFormat;
+  if (!locale) return 'm/d/Y';
+  switch (locale) {
+    case 'de':
+    case 'ru':
+      return 'm.d.Y';
+    case 'it':
+    case 'fr':
+    case 'pt':
+      return 'd/m/Y';
+    case 'zh':
+    case 'zh_tw':
+    case 'ja':
+      return 'Y/m/d';
+    default:
+      return 'm/d/Y';
+  }
+};
```

Comment:
Weighing in; Carbon core doesn't provide locale data due to the maintenance concern (and thus let application deal with it instead as the current `master` does). e.g. Running below code produces some different results from here, e.g. for `en-CA`:

```javascript
console.log('af:', new Intl.DateTimeFormat('af').format(new Date()));
console.log('af-NA:', new Intl.DateTimeFormat('af-NA').format(new Date()));
console.log('af-ZA:', new Intl.DateTimeFormat('af-ZA').format(new Date()));
console.log('agq:', new Intl.DateTimeFormat('agq').format(new Date()));
console.log('agq-CM:', new Intl.DateTimeFormat('agq-CM').format(new Date()));
...
```

Reply:
I can certainly switch it over to that instead of using the hardcoded values.

Reply:
Cool, also note that `m/d/Y` thing is just a default value. Application should take care of putting the right prop value for the locale user is using.